pkg/controllers/federatedresourcequota/federated_resource_quota_sync_controller.go

```diff
@@ -83,7 +83,11 @@ func (c *SyncController) Reconcile(ctx context.Context, req controllerruntime.Re
 		return controllerruntime.Result{}, err
 	}
 
-	if err := c.buildWorks(ctx, quota, clusterList.Items); err != nil {
+	if err := c.cleanUpOrphanWorks(ctx, quota); err != nil {
```

Comment:
Move `cleanUpOrphanWorks` ahead of `Client.List(ctx, clusterList)`, so that if something wrong happened when cleaning the works could save the effort of listing clusters.